plugin/cmd/kube-scheduler/app/server.go

```diff
@@ -18,37 +18,307 @@ limitations under the License.
+	fs.StringVar(&options.config.LeaderElection.LockObjectNamespace, "lock-object-namespace", options.config.LeaderElection.LockObjectNamespace, "Define the namespace of the lock object.")
+	fs.StringVar(&options.config.LeaderElection.LockObjectName, "lock-object-name", options.config.LeaderElection.LockObjectName, "Define the name of the lock object.")
+	fs.IntVar(&options.config.HardPodAffinitySymmetricWeight, "hard-pod-affinity-symmetric-weight", options.config.HardPodAffinitySymmetricWeight,
+		"RequiredDuringScheduling affinity is not symmetric, but there is an implicit PreferredDuringScheduling affinity rule corresponding "+
+			"to every RequiredDuringScheduling affinity rule. --hard-pod-affinity-symmetric-weight represents the weight of implicit PreferredDuringScheduling affinity rule.")
+	fs.MarkDeprecated("hard-pod-affinity-symmetric-weight", "This option was moved to the policy configuration file")
+	fs.StringVar(&options.config.FailureDomains, "failure-domains", options.config.FailureDomains, "Indicate the \"all topologies\" set for an empty topologyKey when it's used for PreferredDuringScheduling pod anti-affinity.")
+	fs.MarkDeprecated("failure-domains", "Doesn't have any effect. Will be removed in future version.")
+	leaderelectionconfig.BindFlags(&options.config.LeaderElection.LeaderElectionConfiguration, fs)
+	utilfeature.DefaultFeatureGate.AddFlag(fs)
+}
+
+func NewOptions() (*Options, error) {
+	o := &Options{
+		config: new(componentconfig.KubeSchedulerConfiguration),
+	}
+
+	o.scheme = runtime.NewScheme()
+	o.codecs = serializer.NewCodecFactory(o.scheme)
+
+	if err := componentconfig.AddToScheme(o.scheme); err != nil {
+		return nil, err
+	}
+	if err := componentconfigv1alpha1.AddToScheme(o.scheme); err != nil {
+		return nil, err
+	}
+
+	return o, nil
+}
+
+func (o *Options) Complete() error {
+	if len(o.ConfigFile) == 0 {
+		glog.Warning("WARNING: all flags than --config are deprecated. Please begin using a config file ASAP.")
+		o.applyDeprecatedHealthzAddressToConfig()
+		o.applyDeprecatedHealthzPortToConfig()
+		o.applyDeprecatedAlgorithmSourceOptionsToConfig()
+	}
+
+	return nil
+}
+
+// applyDeprecatedHealthzAddressToConfig sets o.config.HealthzBindAddress and
+// o.config.MetricsBindAddress from flags passed on the command line based on
+// the following rules:
+//
+// 1. If --address is empty, leave the config as-is.
+// 2. Otherwise, use the value of --address for the address portion of
+//    o.config.HealthzBindAddress
+func (o *Options) applyDeprecatedHealthzAddressToConfig() {
+	if len(o.healthzAddress) == 0 {
+		return
+	}
+
+	index := strings.Index(o.config.HealthzBindAddress, ":")
+	if index != -1 {
+		port := o.config.HealthzBindAddress[:index+1]
+		bindAddr := fmt.Sprintf("%s:%s", o.healthzAddress, port)
```

Comment:
Please use [net.joinHostPort](https://golang.org/pkg/net/#JoinHostPort) for IPv6 compatibility.

Reply:
Fixed